### DISCUSSION TITLE
Specify which folders to include in the .zip that is created and uploaded by the databricks_pyspark_step_launcher to Databricks

### DISCUSSION THREAD

**Discussion by weberdavid** [dagster-io/dagster#17948] [Q&A]
**Context**
We use the `databricks_pyspark_step_launcher` to execute assets on Databricks.
As we also deal with dbt-integrations, and a lot of other code that is not connected to the ingestion jobs running on Databricks, we want to be careful about what we include in the .zip that is generated by the `databricks_pyspark_step_launcher`.

**Problem**
In some instances, the .zip file gets quite big (due to the number of files in the repos), and uploading time to Databricks increases tremendeously. We want to avoid waiting for several minutes just for the code to arrive in Databricks.

**Question / Solution**
There should be a way of specifying, which folders should be included in the .zip process of the `databricks_pyspark_step_launcher`. I investigated a bit, and it seems that [this ](https://github.com/dagster-io/dagster/blob/8978f184b8b041e7f37fbdd886a1800d27fc87a2/python_modules/libraries/dagster-databricks/dagster_databricks/databricks_pyspark_step_launcher.py#L492) is the code where the zipping is executed. The referenced function `build_pyspark_zip` ([here](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-pyspark/dagster_pyspark/utils.py)) seems to take a parameter `exclude`, where you can pass a list of folders to exclude in the zipping. However, this option is not propagated upwards to the `databricks_pyspark_step_launcher`.

If there is no other way to achieve this, I am happy to create an official issue for that and work on a PR that gives this "exclude option" on the `databricks_pyspark_step_launcher` level.

**Comment by Zeulni** [comment]
We have the same issue in our project, a solution would be highly appreciated!

**Comment by sryza** [comment]
Closing this in favor of the corresponding Github Issue: https://github.com/dagster-io/dagster/issues/18099
